### ISSUE TITLE
Example code seems wrong

### ISSUE THREAD

**Issue by Frydac** [rubyzip/rubyzip#377]
I created some functionality to zip a folder recursively based on the example code [here](https://github.com/rubyzip/rubyzip/blob/05916bf89181e1955118fd3ea059f18acac28cc8/samples/example_recursive.rb), however, the files before zipping where not binary equal to the unzipped files.

When I changed the following line:
```
    else
        io.get_output_stream(zipFilePath) { |f| f.puts(File.open(diskFilePath, "rb").read())}
    end
```
to 
```
    else
        io.get_output_stream(zipFilePath) { |f| f.write(File.open(diskFilePath, "rb").read())}
    end
```
Files are binary equal after zipping/unzipping

EDIT: I still had a mistake.. imho which shows the example code should be tested thoroughly on multiple platforms.

**Comment by Frydac** [comment]
On windows I still have troubles with binary equality of some files when I zip a directory with like 80 files, one of which is 25MB+, and it also contains some empty directories.

I can't share any details on the exact files, maybe I'll try to reproduce at some point and provide more information. But for now I'm afraid, I'm giving up on ruby zip.

**Comment by jdleesmiller** [comment]
Thanks for getting in touch. That example code seems quite old. The latest code on master doesn't appear to have the same problem. It looks like this was fixed in #297.
